examples/qtmegachatapi/chatSettings.cpp

```diff
@@ -6,10 +6,47 @@ ChatSettings::ChatSettings(QMainWindow *parent)
     :QDialog(parent),
     ui(new Ui::SettingsDialog)
 {
+    mMainWin = (MainWindow *) parent;
     ui->setupUi(this);
+    #ifndef KARERE_DISABLE_WEBRTC
+        mega::MegaStringList *audioInDevices = mMainWin->mMegaChatApi->getChatAudioInDevices();
+        for (int i=0; i<audioInDevices->size(); i++)
+        {
+            ui->audioInCombo->addItem(audioInDevices->get(i));
+        }
+        mAudioInIdx = ui->audioInCombo->currentIndex();
+        delete audioInDevices;
+
+        mega::MegaStringList *videoInDevices = mMainWin->mMegaChatApi->getChatVideoInDevices();
+        for (int i=0; i<videoInDevices->size(); i++)
+        {
+            ui->videoInCombo->addItem(videoInDevices->get(i));
+        }
+        mVideoInIdx = ui->videoInCombo->currentIndex();
+        delete videoInDevices;
+    #endif
 }
 
 ChatSettings::~ChatSettings()
 {
     delete ui;
 }
+
+void ChatSettings::on_buttonBox_clicked(QAbstractButton *button)
+{
+    #ifndef KARERE_DISABLE_WEBRTC
+        if (ui->audioInCombo->currentIndex() != mAudioInIdx)
+        {
+            mAudioInIdx = ui->audioInCombo->currentIndex();
+            std::string device =  ui->audioInCombo->itemText(ui->audioInCombo->currentIndex()).toLatin1().data();
+            mMainWin->mMegaChatApi->setChatAudioInDevice(device.c_str());
```

Comment:
Maybe we should add a check here. If `setChatAudioInDevice()` returns false, it means it has failed